backend/app/services/implementations/user_service.py

```diff
@@ -0,0 +1,88 @@
+import logging
+
+import firebase_admin.auth
+from fastapi import HTTPException
+from sqlalchemy.orm import Session
+
+from app.models import User
+from app.schemas.user import UserCreate, UserInDB, UserRole
+from app.services.interfaces.user_service import IUserService
+
+
+class UserService(IUserService):
+    def __init__(self, db: Session):
+        self.db = db
+        self.logger = logging.getLogger(__name__)
+
+    async def create_user(self, user: UserCreate) -> UserInDB:
+        firebase_user = None
+        try:
+            # Create user in Firebase
+            firebase_user = firebase_admin.auth.create_user(
+                email=user.email, password=user.password
+            )
+
+            role_id = UserRole.to_role_id(user.role)
+
+            # create user in database
+            db_user = User(
+                first_name=user.first_name,
+                last_name=user.last_name,
+                email=user.email,
+                role_id=role_id,
+                auth_id=firebase_user.uid,
+            )
+
+            self.db.add(db_user)
+            self.db.commit()
```

Comment:
```suggestion
            self.db.commit() # Finish database transaction and run previously defined database operations (ie. db.add)
```
Adding comments to db calls to provide more context for other readers, this is more than typically necessarily